### ISSUE TITLE
Wrong HTTP request code generation

### ISSUE THREAD

**Issue by chimit** [a85/POSTMan-Chrome-Extension#5585]
This request:

![image](https://user-images.githubusercontent.com/839349/49207336-95c13280-f3ef-11e8-99f7-91cd82d8e44c.png)

Generates this HTTP code:

![image](https://user-images.githubusercontent.com/839349/49207374-b4bfc480-f3ef-11e8-80be-430510c0115d.png)

But should be:
```http
POST  HTTP/1.1
Host: localhost
Authorization: Bearer abc
Content-Type: application/x-www-form-urlencoded
cache-control: no-cache

name=John&last_name=Doe&foo=bar
```

Problems:

1. No ampersand separators between key-value pairs in the body;
2. `undefined=undefined`?
3. AFAIK body should be separated from the header params by two `\n`, not one.

**Comment by StevenStreib** [comment]
I can replicate the issue exactly as described. Just as an additional detail, the code generation appears to be broken in a few different ways (undefined key with no value, undefined key with undefined value, missing ampersands, etc.) depending on the selected language.
The only two generators that appear to be unaffected by this are:
- Javascript Jquery AJAX
- NodeJS Unirest

**Comment by a85** [comment]
@numaanashraf We should probably do a validation step and show warnings in the code generator before we send it for parsing. @vkaegis Tagging you here.

**Comment by stephenmee** [comment]
Is there any workaround for this issue?  Is there a ETA for a fix?  Thanks.

**Comment by jun488** [comment]
We use Postman code snippets extensively in our ticketing system. HTTP code is great as it is very short, but this bug makes it useless for most scenarios unfortunately.

Compare HTTP: `searchName=Test+searchsearch%5BidxID%5D=a002search%5Bhp%5D=100000undefined=undefined`

vs cURL" `  -d 'searchName=Test%20search&search%5BidxID%5D=a002&search%5Bhp%5D=100000&undefined='`

Jquery AJAX: `  "data": {
    "searchName": "Test search",
    "search[idxID]": "a002",
    "search[hp]": "100000"
  }`

**Comment by mccannt** [comment]
Just investigated this issue as well with some of my team. Looks like its impacting some code snippets but not all. 

Looks to be a visual issue but can propagate if someone is to share the Code Snippet with others and they import/run the calls with this undefined parameter.

**Comment by vinitshahdeo** [comment]
I tried to reproduce **_(App version 7.0.9)_** this but I'm getting the correct code snippet.

<img width="1172" alt="Screenshot 2019-04-30 at 2 22 33 PM" src="https://user-images.githubusercontent.com/20594326/56950715-bf73b680-6b53-11e9-9961-2628bb5155c3.png">

<img width="721" alt="Screenshot 2019-04-30 at 2 25 50 PM" src="https://user-images.githubusercontent.com/20594326/56950870-19747c00-6b54-11e9-859a-b826252b90d1.png">

@chimit @mccannt 
Can you please confirm from your end if I'm missing something?

**Comment by mccannt** [comment]
@vinitshahdeo I am not able to replicate this now using 7.0.9. 
However I have a team member using 7.0.7 and he was able to re-produce it using that a couple of weeks back. 

I am not convinced that its resolved, there is a just a specific scenario that needs to happen I believe. These things never fix themselves and if they do .... RUN! :P

**Comment by abhijitkane** [comment]
@chimit - Not able to reproduce this anymore. Also, in Postman 7.2.1, we've introduced bug-fixes, support for two new languages (JS-Fetch and Powershell), and the ability to configure certain aspects of the generated snippet with the 'Experimental Codegen Mode' flag enabled in Postman's global settings.

Closing this for now, but do feel free to add a comment in case you see this issue with the 'Experimental Codegen Mode' flag enabled in v7.2.1.
